### PR TITLE
Add search suggest + small fixes

### PR DESCRIPTION
**Pull request by kusem**
Improve Search
Add search suggest
Fix front

### CODE REVIEW

File: src/Controller/SearchController.php [hunk 1]
```diff
@@ -2,16 +2,39 @@
...
+     * @return JsonResponse
+     */
+    #[Route('/quick_search', name: 'quick_search')]
+    public function quickSearch(Request $request, SearchesRepository $search)
+    {
+        $wordBegin = $request->query->get('wordBegin');
+        if(strlen($wordBegin)>2){
+            return  new JsonResponse([
+                'status'    => 1,
+                'matches'   => array_column((array) $search->getFastSearchMatchWords($wordBegin), 'word')
```
Comment (TheVitik) [inline src/Controller/SearchController.php]: Make a function name shorter.
Function should return array with words, so remove array_column function and type casting.

File: src/Controller/SearchController.php [hunk 2]
```diff
@@ -2,16 +2,39 @@
...
+     * @param Request $request
+     * @param SearchesRepository $search
+     * @return JsonResponse
+     */
+    #[Route('/quick_search', name: 'quick_search')]
+    public function quickSearch(Request $request, SearchesRepository $search)
+    {
+        $wordBegin = $request->query->get('wordBegin');
+        if(strlen($wordBegin)>2){
+            return  new JsonResponse([
```
Comment (TheVitik) [inline src/Controller/SearchController.php]: Remove space before `new JsonResponse`

File: src/Controller/SearchController.php [hunk 3]
```diff
@@ -2,16 +2,39 @@
...
 {
+    /**
+     * Returns array of possible words with same beginning
+     *
+     * @param Request $request
+     * @param SearchesRepository $search
+     * @return JsonResponse
+     */
+    #[Route('/quick_search', name: 'quick_search')]
+    public function quickSearch(Request $request, SearchesRepository $search)
```
Comment (TheVitik) [inline src/Controller/SearchController.php]: Add return type declaration `JsonResponse`

File: src/Repository/SearchesRepository.php [hunk 4]
```diff
@@ -35,6 +35,22 @@ public function getTopTags(): array
             ->getResult();
     }
 
+    /**
+     * @return Searches[] Returns an array of top searched words with same beggining
+     */
+    public function getFastSearchMatchWords($wordBegin): array
```
Comment (TheVitik) [inline src/Repository/SearchesRepository.php:42]: Add type hint

File: templates/index.html.twig [hunk 5]
```diff
@@ -7,10 +7,44 @@
...
+                        $("#search").autocomplete("option", {
+                            source: availableWords
+                        });
+                    } else {
+                        availableWords = [];
+                    }
+                });
+            }
+
+            document.addEventListener("DOMContentLoaded", function () {
```
Comment (TheVitik) [inline templates/index.html.twig]: May be better to use `$(document).ready(function(){` ?

File: src/Controller/SearchController.php [hunk 6]
```diff
@@ -2,16 +2,39 @@
...
+    {
+        $wordBegin = $request->query->get('wordBegin');
+        if(strlen($wordBegin)>2){
```
Comment (taras-koval) [inline src/Controller/SearchController.php]: add spaces after if and between ">"

File: src/Controller/SearchController.php (hunk 1 above)
Comment (kusem) [reply src/Controller/SearchController.php]: Function name fully represent what it does - so I believe it's ok.
Type casting remove - ok
array_column should be here, it's a part of the algorithm.

File: templates/index.html.twig (hunk 5 above)
Comment (kusem) [reply templates/index.html.twig]: > 

It will not work - webpack loads last, so jquery will not work.

File: src/Controller/SearchController.php [hunk 7]
```diff
@@ -2,16 +2,39 @@
...
+            return  new JsonResponse([
+                'status'    => 1,
+                'matches'   => array_column((array) $search->getFastSearchMatchWords($wordBegin), 'word')
+            ]);
+        }
+
+        return new JsonResponse(['status'    => 0]);
```
Comment (TheVitik) [inline src/Controller/SearchController.php]: Remove funny spaces

File: src/Controller/SearchController.php (hunk 6 above)
Comment (TheVitik) [reply src/Controller/SearchController.php]: And space after `if` and before `{` ☺

File: assets/app.js [hunk 8]
```diff
@@ -5,39 +5,23 @@
...
 
-//import '~/bootstrap/scss/bootstrap.scss';
 const $ = require('jquery');
 global.$ = global.jQuery = $;
 
-//am4art
-/*import './js/amcharts/core';
-import './js/amcharts/charts';
-import './js/amcharts/wordCloud';
-import './js/amcharts/moonrisekingdom';
```
Comment (yderevianko-cartera) [inline assets/app.js:23]: why did you delete charts ?

File: src/Controller/SearchController.php [hunk 9]
```diff
@@ -2,16 +2,43 @@
...
+            return new JsonResponse([
+                'status'    => 1,
+                'matches'   => $search->getFastSearchMatchWords($wordBegin)
+            ]);
+        }
+
+        return new JsonResponse([
+            'status' => 0,
+            'message'=> 'At least 3 symbols required for search initiation.'
+            ]);
```
Comment (yderevianko-cartera) [inline src/Controller/SearchController.php]: ]); should be on same level as return

File: templates/index.html.twig [hunk 10]
```diff
@@ -7,10 +7,44 @@
...
                 </form>
             </div>
         </div>
+
         {% include 'blocks/_tag_cloud.html.twig' %}
+
+        <script>
+            let availableWords = [];
+
+            function getPossibleWords(wordBegin) {
```
Comment (yderevianko-cartera) [inline templates/index.html.twig]: please move js to separate file and inject it using webpack

File: templates/search.html.twig [hunk 11]
```diff
@@ -1,33 +1,151 @@
...
+            <button class="btn btn-primary">Search</button>
         </form>
-        <h3 class="text-primary mt-4">{{ word | title }}</h3>
+
+        <script>
+            let availableWords = [];
+
+            function getPossibleWords(wordBegin) {
+                $.get("/quick_search?wordBegin=" + wordBegin, function (data) {
+                    if (data.status === 1) {
```
Comment (yderevianko-cartera) [inline templates/search.html.twig]: please move js to separate file and inject it using webpack

File: templates/search.html.twig [hunk 12]
```diff
@@ -1,33 +1,151 @@
...
+            {% for pronunciation in entry.pronunciations %}
+            <div id="player_raw" style="display: none;">
+                <audio class="d-block mt-1" controls>
+                    <source src="{{ pronunciation }}" type="audio/mpeg">
+                    Your browser does not support the audio tag.
+                </audio>
+            </div>
+            <script>
+                window.addEventListener('DOMContentLoaded', () => {
+                    $('#player_new').html($("#player_raw").html());
```
Comment (yderevianko-cartera) [inline templates/search.html.twig]: please move js to separate file and inject it using webpack

File: assets/app.js (hunk 8 above)
Comment (kusem) [reply assets/app.js:23]: > why did you delete charts ?

It was COMMENTED rudiment import, before I installed them via NPM

**Review by yderevianko-cartera — APPROVED** [review]
